tests/test_encoding/test_count_frequency_encoder.py

```diff
@@ -122,10 +122,28 @@ def test_error_if_input_df_contains_categories_not_present_in_fit_df(
 ):
     # test case 3: when dataset to be transformed contains categories not present in
     # training dataset
-    with pytest.warns(UserWarning):
-        encoder = CountFrequencyEncoder()
-        encoder.fit(df_enc)
-        encoder.transform(df_enc_rare)
+    msg = "During the encoding, NaN values were introduced in the feature(s) var_A."
+    encoder = CountFrequencyEncoder()
+
+    if encoder.rare_labels == "ignore":
+        with pytest.warns(UserWarning) as record:
+            encoder.fit(df_enc)
+            encoder.transform(df_enc_rare)
+
+        # check that only one warning was raised
+        assert len(record) == 1
+        # check that the message matches
+        assert record[0].message.args[0] == msg
+
```

Comment:
now we need to specify the encoder again but with rare labels raise. Please add

encoder = CountFrequencyEncoder(rare_labels='raise')